utils/polus-filepattern-util/README.md

```diff
@@ -25,31 +45,63 @@ image_c001_z001.ome.tif
+This function retrieves all files that match specific variables values. Using
+the example filenames presented in the **File Pattern Format** section, if `C=0`
+is passed as an input argument, then `get_matching` will return all files that
+contain `_c000`. However, the input values do not need to be a single value,
+they can be a list. So if `C=[0,1]`, then a list of all files will be returned
+such that each file would contain `_c000` or `_c001`.
 
-The list that is returned contains dictionaries. Each dictionary contains a key for each variable parsed from the filename and a `file` key that indicates the name of the file.
+The list that is returned contains dictionaries. Each dictionary contains a key
+for each variable parsed from the filename and a `file` key that indicates the
+name of the file.
 
 ### FilePattern.iterate
 
-This function is an iterable that returns a list of filenames every time it is called. Each call returns a list of files that match a unique combination of variable values so that every image that matches a file pattern is contained in only one of the calls to this function. Specifying a list of variables in the `group_by` parameter will return a list of filenames with all variable values constant except those indicated by `group_by`. Using the example filenames presented in the **File Pattern Format** section, if `group_by='z'` then the top three files are returned by the first call to this function and the bottom three files are returned by the second call to this function.
+This function is an iterable that returns a list of filenames every time it is
+called. Each call returns a list of files that match a unique combination of
+variable values so that every image that matches a file pattern is contained in
+only one of the calls to this function. Specifying a list of variables in the
+`group_by` parameter will return a list of filenames with all variable values
+constant except those indicated by `group_by`. Using the example filenames
+presented in the **File Pattern Format** section, if `group_by='z'` then the top
+three files are returned by the first call to this function and the bottom three
+files are returned by the second call to this function.
 
-In addition to the `group_by` argument, it is possible to pass arguments matching the `get_matching` function. This will cause `iterate` to only return files matching specific variables values.
+In addition to the `group_by` argument, it is possible to pass arguments
+matching the `get_matching` function. This will cause `iterate` to only return
```

Comment:
This is a comment for your usage pattern. Since you define `__iter__` won't it be cleaner if you iterate over the variable itself rather than call `.iterate()` ?

Reply:
This was resolved by removing the old information from the README.